request/quic_transport.go

```diff
@@ -1,26 +1,34 @@
 package request
 
 import (
+	"log"
 	"net/http"
 	"strings"
+	"sync"
 	"sync/atomic"
 
 	"github.com/NordSecurity/nordvpn-linux/events"
+	"github.com/NordSecurity/nordvpn-linux/internal"
 	"github.com/quic-go/quic-go/http3"
 )
 
 // QuicTransport is modified/enhanced RoundTripper.
 // Thread safe.
 type QuicTransport struct {
-	inner    *atomic.Pointer[http3.RoundTripper]
-	createFn func() *http3.RoundTripper
+	// for protects the access to inner and to shouldRecreate.
```

Comment:
Comment language should be reviewed. Last part says that lock is needed to protect atomic variable?

Reply:
yeah, it is correct.

The atomic is set to `true` when a request returns an error. In that case the RWMutex is locked for `read`.
Then the atomic is set to `false` when `inner` is recreated, in which case the mutex is locked for `writing`. So nobody can change the value of the atomic unless it has the lock in read or write. It cannot be boolean because there are multiple "readers" that can change its value in the same time.